precompile/allowlist/config.go

```diff
@@ -14,18 +14,26 @@ import (
 // AllowListConfig specifies the initial set of addresses with Admin or Enabled roles.
 type AllowListConfig struct {
 	AdminAddresses   []common.Address `json:"adminAddresses,omitempty"`   // initial admin addresses
+	ManagerAddresses []common.Address `json:"managerAddresses,omitempty"` // initial manager addresses
 	EnabledAddresses []common.Address `json:"enabledAddresses,omitempty"` // initial enabled addresses
 }
 
 // Configure initializes the address space of [precompileAddr] by initializing the role of each of
 // the addresses in [AllowListAdmins].
-func (c *AllowListConfig) Configure(state contract.StateDB, precompileAddr common.Address) error {
+func (c *AllowListConfig) Configure(chainConfig precompileconfig.ChainConfig, precompileAddr common.Address, state contract.StateDB, blockContext contract.BlockContext) error {
 	for _, enabledAddr := range c.EnabledAddresses {
 		SetAllowListRole(state, precompileAddr, enabledAddr, EnabledRole)
 	}
 	for _, adminAddr := range c.AdminAddresses {
 		SetAllowListRole(state, precompileAddr, adminAddr, AdminRole)
 	}
+	// Verify() should have been called before Configure()
+	// so we know manager role is activated
+	if chainConfig.IsDUpgrade(blockContext.Timestamp()) {
```

Comment:
If this comment is true, then why do we need this check here?

Reply:
kept it since we already exposed block context to Configure. removed now.